src/nodes.coffee

```diff
@@ -1810,7 +1821,7 @@ exports.Assign = class Assign extends Base
     answer = compiledName.concat @makeCode(" #{ @context or '=' } "), val
     # Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assignment_without_declaration,
     # if we’re destructuring without declaring, the destructuring assignment must be wrapped in parentheses.
-    if o.level > LEVEL_LIST or (isValue and @variable.base instanceof Obj and not @param)
+    if o.level > LEVEL_LIST or (isValue and @variable.base instanceof Obj and not @nestedLhs and not @param)
```

Comment:
@connec nope it seems to interfere w/ the wrapping in parens of top-level destructured object assigns. Not crystal clear on the different nested `Obj`/`Assign` structures so there may be a way do it without introducing `nestedLhs` but that's the way I found to distinguish the top-level `Assign`s from the nested ones